lib/constants.js

```diff
@@ -430,6 +430,12 @@ export const LANGUAGES = [
     mime: 'application/typescript',
     short: 'typescript'
   },
+  {
+    name: 'TSX',
+    mode: 'jsx',
+    mime: 'text/typescript-jsx',
```

Comment:
I'm fairly certain you do not need `mime` field here. Please remove 👍

Reply:
I'm not sure on the verbage exactly, but it seems to distinguish `text/jsx` and `text/typescript-jsx` MIME types within the single `JSX` mode. 

https://codemirror.net/mode/jsx/index.html